### PR TITLE
chore: add ts-jest testing config and scripts

### PR DESCRIPTION
**Pull request by kotarella1110**
The current tests are not run properly.
The cause of tests not running is removed `ts-jest` by [this commit](https://github.com/thebrodmann/deox/commit/e185e358014b03905c861aef7aa5576bd38c32fe#diff-2d0cd5d10b9604941c38c6aac608178a).

Both jest-runner-tsc and ts-jest are required.
Please see below for the details.

https://github.com/azz/jest-runner-tsc/issues/6#issuecomment-395934357

> They are different things :) ts-jest is to transform TS code into JS. jest-runner-tsc doesn't run tests, it runs the TS typechecker, so it's like a linter. You need to run both.

### CODE REVIEW

**Comment by kotarella1110** [comment]
Once this pull request is merged, I can create a PR for #87 !

File: package.json [hunk 1]
```diff
@@ -83,7 +86,7 @@
   "husky": {
     "hooks": {
       "pre-commit": "lint-staged",
-      "pre-push": "npm run lint && npm run test -- --bail && npm run test-dts -- --bail",
+      "pre-push": "npm run lint && npm run test -- --bail && npm run test-tsc -- --bail && npm run test-dts -- --bail",
```
Comment (thebrodmann) [inline package.json]: I think type checking needs fewer resources than unit testing. So we can run type checker before unit tests, in this way if there be a type error in tests, the push process will fail earlier without running extra tests.

File: package.json [hunk 2]
```diff
@@ -92,6 +95,7 @@
       "npm run lint:fix --",
       "git add",
       "npm run test -- --bail --findRelatedTests",
+      "npm run test-tsc -- --bail --findRelatedTests",
```
Comment (thebrodmann) [inline package.json]: So here.

File: .circleci/config.yml [hunk 3]
```diff
@@ -30,6 +30,10 @@ jobs:
           name: Unit Testing
           command: yarn run test --ci --coverage
 
+      - run:
+          name: TSC Testing
+          command: yarn run test-tsc --ci
+
```
Comment (thebrodmann) [inline .circleci/config.yml]: So here.

**Comment by kotarella1110** [comment]
completed 👍

**Comment by thebrodmann** [comment]
Merged with honor @kotarella1110.

**Pull request merged**